### PR TITLE
wastfmt: tool to reformat and desugar a wast file

### PR DESCRIPTION
**Pull request by binji**

### CODE REVIEW

**Comment by binji** [comment]
@dschuff

File: test/wastfmt/basic.txt [hunk 1]
```diff
@@ -0,0 +1,27 @@
+;;; TOOL: wastfmt
+(module
+  (import "foo" "bar" (func (result i32)))
```
Comment (dschuff) [inline test/wastfmt/basic.txt]: I guess we'd need separate files to test for imported memory and tables?

File: CMakeLists.txt [hunk 2]
```diff
@@ -264,6 +253,28 @@ if (NOT EMSCRIPTEN)
     target_link_libraries(wasm-interp m)
   endif ()
 
+  # wastfmt
+  add_executable(wastfmt src/exe/wastfmt.c)
```
Comment (binji) [inline CMakeLists.txt]: OK, I was trying to think of something clever for removing sugar (`nutra-wast`, `wast-zero`, `diet-wast`). But maybe just `wast-desugar` or `wast-canonicalize`?

File: test/wastfmt/basic.txt (hunk 1 above)
Comment (binji) [inline test/wastfmt/basic.txt]: Yeah, I just wanted to add a basic test here to make sure things are working, adding more would be good though.

File: CMakeLists.txt (hunk 2 above)
Comment (dschuff) [inline CMakeLists.txt]: Either of those would probably be OK with me. Maybe we should pick something that reflects the same language used where either the spec or design doc (when it exists) talks about sugar, or macros, or whatever. It looks like the spec just refers to "sugar" so maybe `wast-desugar` for now. When we introduce some text somewhere that describes the text format and its sugar, if we decide on a better term there, we can rename this to match.

**Comment by binji** [comment]
true, true. I've been pretty good about writing tests, though :)

**Comment by dschuff** [comment]
Yup; hence the LGTM :)

**Pull request closed without merging**